pandas/indexes/base.py

```diff
@@ -1745,29 +1745,13 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if is_list_like(other):
-            warnings.warn("using '+' to provide set union with Indexes is "
-                          "deprecated, use '|' or .union()", FutureWarning,
-                          stacklevel=2)
-        if isinstance(other, Index):
```

Comment:
I think we are expanding the api w/o any testing of ``__add__``. Its ok, but should create an issue to add tests and/or limit it. (so ideally we don't expand it at all for ``Index`` and add back if needed later), esp NOT for ``MultiIndex`` and ``Categorical`` (I think you just renamed, but just to be sure).